lib/timetable-form/FormStage.tsx

```diff
@@ -0,0 +1,93 @@
+import { ReactNode, useState } from "react";
+
+import { EventKey } from "../constants";
+import { Button, DateInput, ErrorSummary, TextArea } from "../gds-components";
+import { useValidation } from "./useValidation";
+
+import styles from "./styles.module.css";
+import "govuk-frontend/dist/govuk/govuk-frontend.min.css";
+
+interface FormStageProps {
+  title: string;
+  description: ReactNode;
+  startEvent: EventKey;
+  endEvent?: EventKey;
+}
+
+export const FormStage = ({
+  title,
+  description,
+  startEvent,
+  endEvent,
+}: FormStageProps): JSX.Element => {
+  const [startDate, setStartDate] = useState<string>("");
+  const [endDate, setEndDate] = useState<string>("");
+  const [additionalInformation, setAdditionalInformation] =
+    useState<string>("");
+
+  const { errors, validateDevelopmentPlanEventDates } = useValidation();
+
+  const handleValidate = () => {
+    const stageEventsDates = [{ reference: startEvent, eventDate: startDate }];
+
+    if (endEvent) {
+      stageEventsDates.push({ reference: endEvent, eventDate: endDate });
+    }
+
+    validateDevelopmentPlanEventDates(stageEventsDates);
+  };
+
+  return (
+    <div>
+      <div className={styles.form}>
+        <a href="#" className="govuk-back-link">
+          Back
+        </a>
+        <h1 className="govuk-heading-l" data-testid="form-title">
+          {title} Stage
+          <span className="govuk-caption-m govuk-!-margin-top-2">
+            {description}
+          </span>
+        </h1>
+        <ErrorSummary errors={errors} />
+        <div id={`${startEvent}-eventDate`}>
+          <DateInput
+            value={startDate}
+            label={endEvent ? "Start Date" : "Date"}
+            name={`${startEvent}-start-date`}
+            error={
+              errors.find(
+                (error) =>
+                  error.path[0] === startEvent && error.path[1] === "eventDate"
+              )?.message
+            }
+            onChange={setStartDate}
+          />
+        </div>
+        {endEvent && (
+          <div id={`${endEvent}-eventDate`}>
+            <DateInput
+              value={endDate}
+              label="End Date"
+              name={`${endEvent}-end-date`}
+              error={
+                errors.find(
+                  (error) =>
+                    error.path[0] === endEvent && error.path[1] === "eventDate"
+                )?.message
+              }
+              onChange={setEndDate}
+            />
+          </div>
+        )}
+        <TextArea
+          label="Additional Information (optional)"
+          onChange={setAdditionalInformation}
+          value={additionalInformation}
+          hint="You can enter up to 100 characters"
+        />
+        <Button onClick={handleValidate}>Continue</Button>
```

Comment:
So is this a decision that Back and Continue will live inside each stage component instead of in the wrapper?

Reply:
the way I see it they are inside this component only momentarily to showcase the validation is still working. I would like to create a container component (i.e.FormPage) and the wrapper component as part of a new ticket to followup this one as I think that will prompt the need for centralised state management. Lemme know your thoughts

Reply:
OK, I think that makes sense, we can figure out where this will actually live later